coralproject/talk
# [CORL-3180]: update to not autoplay tenor gifs in mod queue

## What does this PR do?

These changes update so that Tenor gifs are not autoplaying in the mod queue. For Tenor gifs, we are now retrieving the `mp4` and `gifpreview` formats from the Tenor api and attaching those to the media embed on the comment, so that they can be used to play/pause the gifs in the mod queue, similar to how Giphy gifs display if enabled.

To support any Tenor gifs that have been added since Tenor has been an optional and prior to these changes, this adds a fallback of show/hide Gif, similar to what is seen for Gifs in the comment stream.

I also moved Tenor types out into `common` where Giphy types currently live for consistency and so that we can use them in different places.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

These changes add `still`, `video`, `width`, and `height` to `TenorMedia` schema, to make it consistent with what Giphy has to enable it to be play/paused in the mod queue.

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can have Tenor enabled for gifs, and add some comments with gifs to the comment stream. Go to the admin and find the comments you added in the moderation queue. See that you can play/pause the gifs and that they are not autoplaying.

Add comments with Tenor gifs on `develop` or another branch without these changes, and see that these gifs (when back on this branch) include a show/hide Gif button that can be toggled. This provides a fallback for any comments with Tenor gifs that don't have a still/video to use for play/pause.

## Were any tests migrated to React Testing Library?



## How do we deploy this PR?
